streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -83,7 +89,38 @@ public SubscriptionInfo(final int version,
         if (version >= 3) {
             data.setLatestSupportedVersion(latestSupportedVersion);
         }
-        data.setProcessId(processId);
+        if (version >= 7) {
+            setTaskOffsetSumDataFromTaskOffsetSumMap(data, taskOffsetSums);
+        } else {
+            setPrevAndStandbySetsFromParsedTaskOffsetSumMap(data, taskOffsetSums);
+        }
+        this.data = data;
+    }
+
+    private static void setTaskOffsetSumDataFromTaskOffsetSumMap(final SubscriptionInfoData data,
+                                                                 final Map<TaskId, Integer> taskOffsetSums) {
+        data.setTaskOffsetSums(taskOffsetSums.entrySet().stream().map(t -> {
+            final SubscriptionInfoData.TaskOffsetSum taskOffsetSum = new SubscriptionInfoData.TaskOffsetSum();
+            taskOffsetSum.setTopicGroupId(t.getKey().topicGroupId);
+            taskOffsetSum.setPartition(t.getKey().partition);
+            taskOffsetSum.setOffsetSum(t.getValue());
+            return taskOffsetSum;
+        }).collect(Collectors.toList()));
+    }
+
+    private static void setPrevAndStandbySetsFromParsedTaskOffsetSumMap(final SubscriptionInfoData data,
+                                                                        final Map<TaskId, Integer> taskOffsetSums) {
+        final Set<TaskId> prevTasks = new HashSet<>();
+        final Set<TaskId> standbyTasks = new HashSet<>();
+
+        for (final Map.Entry<TaskId, Integer> taskOffsetSum : taskOffsetSums.entrySet()) {
+            if (taskOffsetSum.getValue() == -1) {
```

Comment:
Oh, actually, here's the reason a constant sentinel is nice, but we didn't actually use it!

Reply:
🤦‍♀